tablo/csv_utils.py

```diff
@@ -38,15 +42,37 @@ def prepare_csv_rows(csv_file):
 
     DateType.formats = create_date_formats(day_first=False)
 
-    # We are never wanting boolean types, so remove that from the default list
-    eligible_types = [StringType, DecimalType, IntegerType, DateType]
-    types = type_guess(row_set.sample, types=eligible_types, strict=True)
+    if csv_info:
+        # If we have csv_info, then we know the column types, no need to guess
+        types = types_from_config(row_set.sample, csv_info)
+    else:
+        types = type_guess(row_set.sample, strict=True)
 
     row_set.register_processor(types_processor(types))
 
     return row_set
 
 
+def types_from_config(rows, csv_info):
+    """
+    Determines the type of the columns based on the csvInfo.dataTypes. This allows us to keep from guessing
+    when the sender of the file knows more about the types than we do.
+    """
+    columns = []
+    convert_type = {
+        'integer': IntegerType,
+        'string': StringType,
+        'decimal': DecimalType,
+        'date': DateType,
+        'empty': StringType  # Sender doesn't know type either, default to String
+    }
+    for row in rows:
```

Comment:
Only a suggestion, but this might be simplified as follows:
```python
    for cell in (row for row in rows):
        name_index = csv_info['fieldNames'].index(cell.column.lower())
        columns.append(convert_type[csv_info['dataTypes'][name_index].lower()]())
```